### ISSUE TITLE
Tables are inaccessible on small viewports by voiceover

### ISSUE THREAD

**Issue by jessiehuff** [patternfly/patternfly-next#2540]
Through testing, I noticed our doc tables [are not readable by voiceover at 768px viewport size](https://codesandbox.io/s/2mm0c) and appear empty with 0 columns and 0 rows:

<img width="539" alt="Screen Shot 2019-12-17 at 1 56 52 PM" src="https://user-images.githubusercontent.com/28281340/71025627-6c944280-20d5-11ea-9d40-e8bb7eb322ae.png">

It seems that when table was created there was discussion about using `role=grid` and it's impact on accessibility which may be causing this issue. (https://github.com/patternfly/patternfly-next/pull/749) 

It might be worth stepping back and looking at whether combining a non-table layout with the semantics of a `<table>` element is the right approach.

**Comment by christiemolloy** [comment]
Been investigating the cause of this on https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Table_Role

**The reason this issue is happening is because the table is display grid:**
"Changing the display value of a <table> element to block, grid, or flex will alter its representation in the accessibility tree. This will cause the table to no longer be announced properly by screen reading technology."

**Mozilla makes recommendations for how the responsive table can be read by a screenreader:**
"Best practices
Only use table, tbody, thead, tr, th, td, etc., for data table structure. You can add these ARIA roles to ensure accessibility should the native semantics of the table be removed, such as with CSS. A relevant use case for the ARIA table role is when CSS's display property overrides the native semantics of a table, such as by display: grid. In this case, you can use the ARIA table roles to re-add the semantics."

**Aria table roles:**
- role="rowgroup": An optional child of the table, the row group encapsulates a group of rows, similar to thead, tbody, and tfoot.
- role="row": A row within the table, and optionally within a rowgroup, that is a container for one or more cells, columnheaders, or rowheaders.
- aria-describedby attribute: Takes as its value the id of the element that serves as a description for the table.
- aria-label attribute: The aria-label provides an accessible name for the table.
- aria-colcount attribute: This attribute is only required if the columns are not present in the DOM all the time. It provides an explicit indication of the number of columns in the full table. Set the value to the total number of columns in the full table. If unknown, set aria-colcount="-1".
- aria-rowcount attribute
This attribute is only required if the rows are not present in the DOM all the time, such as scrollable tables that reuse rows to minimize the number of DOM nodes. It provides an explicit indication of the number of rows in the full table. Set the value to the total number of rows in the full table. If unknown, set aria-colcount="-1".
